src/nrniv/neuronapi.cpp

```diff
@@ -0,0 +1,466 @@
+#include "../../nrnconf.h"
+#include "nrniv_mf.h"
+#include "nrnmpi.h"
+#include "nrnmpiuse.h"
+#include "ocfunc.h"
+#include "ocjump.h"
+
+#include "parse.hpp"
+
+#include "hocdec.h"
+#include "section.h"
+
+
+// we define these here to allow the API to be independent of internal details
+typedef enum {
+    STACK_IS_STR = 1,
+    STACK_IS_VAR = 2,
+    STACK_IS_NUM = 3,
+    STACK_IS_OBJVAR = 4,
+    STACK_IS_OBJTMP = 5,
+    STACK_IS_USERINT = 6,
+    STACK_IS_SYM = 7,
+    STACK_IS_OBJUNREF = 8,
+    STACK_UNKNOWN = -1
+} stack_types_t;
+
+typedef hoc_Item nrn_Item;
+
+
+class SectionListIterator {
+  public:
+    SectionListIterator(nrn_Item*);
+    Section* next(void);
+    int done(void);
+
+  private:
+    nrn_Item* initial;
+    nrn_Item* current;
+};
+
+class SymbolTableIterator {
+  public:
+    SymbolTableIterator(Symlist*);
+    char const* next(void);
+    int done(void);
+
+  private:
+    Symbol* current;
+};
+
+/****************************************
+ * Connections to the rest of NEURON
+ ****************************************/
+extern int nrn_nobanner_;
+extern int diam_changed;
+extern int nrn_try_catch_nest_depth;
+extern "C" void nrnpy_set_pr_etal(int (*cbpr_stdoe)(int, char*), int (*cbpass)());
+int ivocmain_session(int, const char**, const char**, int start_session);
+void simpleconnectsection();
+extern Object* hoc_newobj1(Symbol*, int);
+extern void nrn_change_nseg(Section*, int);
+
+extern "C" {
+
+/****************************************
+ * Initialization
+ ****************************************/
+
+int nrn_init(int argc, const char** argv) {
+    nrn_nobanner_ = 1;
+    int exit_status = ivocmain_session(argc, argv, nullptr, 0);
```

Comment:
For sure we should add your API tests to the test suite, so that the coverage report is meaningful.

Reply:
Agree. As I commented on Wednesday, I don't know how to do this though, because it's not a unit test, and it's not a Python test. If you do, please submit a pull request to the `api` branch. Thanks!

Reply:
See #2365.